### PR TITLE
Use state-api for job driver logs

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
Update Actors page with new IA look and feel
Fix actors page IA for job actors

Follow-up PR: Use state-api for actor logs
Follow-up PR 2: Use state-api for serve replica and controller logs.

Actor list page:
Screenshot 2023-05-10 at 2 46 07 PM

Actor detail page:
Screenshot 2023-05-10 at 2 46 03 PM

Job actor detail page:
Screenshot 2023-05-10 at 2 46 31 PM

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
Others LGTM! One nit: This sidebar is for the job not the actor. We should remove it from the actor detail page.

237523443-2d14819a-3cc8-4841-bc86-5fcd4dc6a698

**Comment by alanwguo** [comment]
> 

In the process of trying to get this to work properly, I realize the architecture of the side bar is a bit confusing. Can we save this fix for next time? Let's discuss in person tomorrow. Lot's of details to go over.

**Comment by scottsun94** [comment]
> > 
> 
> In the process of trying to get this side tab stuff to work properly, I realize the architecture of the side bar is a bit confusing. Can we save this fix for next time? Let's discuss in person tomorrow. Lot's of details to go over.

Sounds good. Then LGTM.

File: dashboard/client/src/service/log.ts [hunk 1]
```diff
@@ -49,3 +49,15 @@ export const getLogDetail = async (url: string) => {
 
   return rsp.data as string;
 };
+
+export const getStateApiDownloadLogUrl = (nodeId: string, fileName: string) =>
+  `api/v0/logs/file?node_id=${nodeId}&filename=${fileName}&lines=-1`;
```
Comment (rickyyx) [inline dashboard/client/src/service/log.ts:54]: I am adding support for 
`api/v0/logs/file?job_id=<submit_job_id>&lines=-1` as part of effort to support job log as well. But this also looks fine.

**Review by rickyyx — COMMENTED** [review]
State API endpoints related changes LGTM.

File: dashboard/client/src/service/log.ts [hunk 2]
```diff
@@ -49,3 +49,15 @@ export const getLogDetail = async (url: string) => {
 
   return rsp.data as string;
 };
+
+export const getStateApiDownloadLogUrl = (nodeId: string, fileName: string) =>
```
Comment (galenhwang) [inline dashboard/client/src/service/log.ts:53]: Can we have one file that centralizes all the state API calls? This makes it easier to check which file needs to be changed if our API changes

**Review by galenhwang — APPROVED** [review]

File: dashboard/client/src/service/log.ts (hunk 2 above)
Comment (alanwguo) [reply dashboard/client/src/service/log.ts:53]: I don't really want to change the existing pattern. We currently group by feature, not underlying api implementation.

**Pull request merged**